_drafts/programatically-configured-bots.md

```diff
@@ -52,9 +52,9 @@ robot.alias = '/'
 # like robot name
 robot.adapter.once 'connected', ->
 
-  # require external hubot scripts
-  robot.loadScriptPackage require('hubot-help')
-  robot.loadScriptPackage require('hubot-shipit')
+  # require external hubot script packages
+  robot.loadPackage 'hubot-help'
+  robot.loadPackage 'hubot-shipit'
```

Comment:
> There is already a robot.load function that takes a string path of a script to require and parse the help of. It could be overloaded to also take a function.

I’d move the help text parsing out of the main package, too. Either plugins should provide meta information by themselves or files should be parsed in `bin/hubot` part and passed to the robot constructor. Ideally the main module has no file system interactions at all, it should all happen outside of it, to make it easier to embed in other apps / start programmatically.

The help text parsing is very convenient when it works, but it’s quite fragile (I run into several problems with it when doing the conversion from CoffeeScript to JavaScript). We could e.g. look for help information in the main export of a script, like

```
module.exports = myScript
module.exports.help = {
  description: '...',
  commands: ['...', '...'],
  // ... etc
}

function myScript (robot) {
  // do something
}
```

Developers could put the `help` into its own file, and we could provide a tool that auto-generates / updates that help file for them, it could even be part of an automated release setup. But that way it’s on the script developer to provide help, not on us to parse it on the fly and remain compatible with different syntax over time.

What do you think?